.github/workflows/linting.yml

```diff
@@ -66,16 +66,17 @@ jobs:
       - name: TFLint Modules
         run: cd modules && tflint --recursive -f compact
 
-      - name: TFLint FAST
-        run: |
-          for STAGE_DIR in $(find fast/stages -name main.tf -printf '%h\n' ) ; do
-            STAGE_NAME=$(echo ${STAGE_DIR} | sed -e 's#^fast/stages/\([^/]*\)[/]\?.*$#\1#')
-            VAR_PATH_LEVEL="."
-            if [ -z $(echo ${STAGE_DIR} | sed -e 's#^fast/stages/[^/]*/\(.*\)$##') ] ; then
-              VAR_PATH_LEVEL=".."
-            fi
-            tflint --chdir ${STAGE_DIR} --var-file ${VAR_PATH_LEVEL}/../../../tests/fast/stages/s$(echo ${STAGE_NAME} | sed -e 's/-/_/g')/simple.tfvars
-          done
+      # TODO: re-enable once PR is finalized
+      # - name: TFLint FAST
+      #   run: |
+      #     for STAGE_DIR in $(find fast/stages -name main.tf -printf '%h\n' ) ; do
+      #       STAGE_NAME=$(echo ${STAGE_DIR} | sed -e 's#^fast/stages/\([^/]*\)[/]\?.*$#\1#')
+      #       VAR_PATH_LEVEL="."
+      #       if [ -z $(echo ${STAGE_DIR} | sed -e 's#^fast/stages/[^/]*/\(.*\)$##') ] ; then
+      #         VAR_PATH_LEVEL=".."
+      #       fi
+      #       tflint --chdir ${STAGE_DIR} --var-file ${VAR_PATH_LEVEL}/../../../tests/fast/stages/s$(echo ${STAGE_NAME} | sed -e 's/-/_/g')/simple.tfvars
+      #     done
```

Comment:
I tested locally, and it looks like this is working (and complaining about some stuff)

Reply:
Yep it was giving me errors on the deleted folders, I temporarily commented so as to have the other linting tests here, if you can re-enable plz do :) It's a bit too cryptic for me... maybe it's enough to have it as a shell command in tools and run it from time to time?

Reply:
I re-implemented that in Python and make it 10x more readable.

It still complains about:
```
Warning: [Fixable] List items should be accessed using square brackets (terraform_deprecated_index)

  on fast/stages/1-resman/main.tf line 101:
 101:     ? module.organization.0.tag_keys
 ```
 
And unused `workforce_identity_providers_defs`, which sound legit?

Reply:
it would be great to have the ability to exclude linting checks via file comments, like for most other language linters

Reply:
Proposal: let's not fail the linting checks on the tf linter, but let it continue on failure and post the error in the PR itself? the code to do that is in the fast GH workflows

Reply:
We have annotations defined [here](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/annotations.md), let me try to add the ignore on `workforce_identity_providers_defs `

Reply:
ah great, revert my clumsy tests on the ci file then :)

Reply:
Looks like the silencer doesn't work in this case :-/

Reply:
I reverted the workflow BTW.

Well, if the silencer doesn't work, we need to either disable ftlint for fast, or make it non-blocking. Capturing output and siphoning it to a PR comment on step failure, with ignore errors on the steps might then be the way to go. I can give it a shot, if you find no way to disable that check.

Reply:
Well, tflint is unusable locally for me, I get a ton of errors on required_version attribute, missing version constraints, etc.

I'll disable it for FAST, we can deep dive on how to integrate it later. I don't think a relatively unimportant linting tool should become an impediment to actual work. If that happens, we need to take a serious look at our priorities.

Reply:
You're getting the required_version, and missing version constraints issues, because have to tweak the config a bit. I was supposed to document this in the contributing guide but, as usual, forgot to do it. Sorry about that.

For the record this is what you have to add to ~/.tflint.hcl:
```hcl
plugin "terraform" {
  enabled = true
  preset  = "recommended"
}

rule "terraform_required_providers" {
  enabled = false
}

rule "terraform_required_version" {
  enabled = false
}

```

As for FAST, tflint already detected a few unused variables, unused locals and some other issues. I think it's useful and we should try to keep it. I will take care of documenting it properly if we agree to use it.

Note that we don't have to go overboard with python and more tooling. tflint already supports [problem matchers](
As for FAST, it already detected unused variables, unused locals and some other issues. I think it's usefull and we should try to keep. I will takre care of documenting it properly.), which we can leverage directly from the pipeline to highlight the problem lines.

Reply:
I don't dispute the fact it's a useful tool, I just find it a bit aggressive if it stands in the oath of a large PR. 😀

If we can control it's behaviour via annotations, or make it just output informative messages, I'd love having it.

Let's just keep being mindful so we do not add much more superstructure in the future, or the hurdle to overcome for contributors will get discouraging, and tools end up stealing substantial focus from our real goal
 🤣

Reply:
and in the end annotations work, so all is good in the world :) thanks both for your patience